ibmcloud-powervs: install oras and modify copy-files.sh for image build

Amulyam24 opened the pull request:
Add command installation for oras in prereq.sh and mody copy-files.sh to add missing dependencies.

Fixes: #2365

mkulke commented on src/cloud-api-adaptor/ibmcloud-powervs/image/copy-files.sh:
```diff
@@ -20,13 +21,29 @@ if [ -e ${PODVM_DIR}/files/etc/aa-offline_fs_kbc-resources.json ]; then
 
+if [ -e ${PODVM_DIR}/files/etc/certificates/tls.crt ] && [ -e ${PODVM_DIR}/files/etc/certificates/tls.key ]; then
+    sudo mkdir -p /etc/certificates
+	sudo cp -a ${PODVM_DIR}/files/etc/certificates/tls.crt /etc/certificates/
```
indent

mkulke reviewed (commented):
you might want to quote the template strings, especially the paths "${PODVM_DIR}/...". it might contain a whitespace or other oddities. you can run shellcheck on the script to check for common problems like this

stevenhorsman reviewed (approved):
This looks okay to me, but it would be great if there was a way to de-duplicate some of the copy-file code

ldoktor commented on src/cloud-api-adaptor/ibmcloud-powervs/image/Makefile:
```diff
@@ -10,8 +10,6 @@ export FORWARDER_PORT ?=
 ROOT_DIR := $(dir $(lastword $(MAKEFILE_LIST)))../../
 include $(ROOT_DIR)podvm/Makefile.inc
 
-export PATH := $(PATH):/usr/local/go/bin:$(HOME)/.cargo/bin
```
Yep, it should be up to the user to set the correct PATH, less magic, more documentation :-)

ldoktor commented on src/cloud-api-adaptor/ibmcloud-powervs/image/copy-files.sh:
```diff
@@ -6,27 +6,44 @@ REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/../..
 sudo mkdir -p /etc/containers
-sudo cp /tmp/files/etc/aa-offline_fs_kbc-keys.json /etc/aa-offline_fs_kbc-keys.json
+sudo cp "${PODVM_DIR}"/files/etc/agent-config.toml /etc/agent-config.toml
+sudo cp "${PODVM_DIR}"/files/etc/aa-offline_fs_kbc-keys.json /etc/aa-offline_fs_kbc-keys.json
```
Well IMO quoting the whole string looks nicer (and avoids issues with spaces, if any), but not really a big issue. Maybe consider for the next time...

ldoktor commented on src/cloud-api-adaptor/ibmcloud-powervs/image/copy-files.sh:
```diff
@@ -6,27 +6,44 @@ REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/../..
+
+if [ -e "${PODVM_DIR}"/files/etc/certificates/tls.crt ] && [ -e "${PODVM_DIR}"/files/etc/certificates/tls.key ]; then
+    sudo mkdir -p /etc/certificates
+    sudo cp -a "${PODVM_DIR}"/files/etc/certificates/tls.crt /etc/certificates/
```
No need to use `-a` for individual files, it's useful for recursive or multiple files, but for individual files pure `cp` would do. (also you might simplify it to `sudo cp "${PODVM_DIR}"/files/etc/certificates/tls.{key,crt} /etc/certificates/`)

ldoktor commented on src/cloud-api-adaptor/ibmcloud-powervs/image/copy-files.sh:
```diff
@@ -6,27 +6,44 @@ REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/../..
 PODVM_DIR=${REPO_ROOT}/podvm
 
 sudo mkdir -p /etc/containers
-sudo cp /tmp/files/etc/aa-offline_fs_kbc-keys.json /etc/aa-offline_fs_kbc-keys.json
```
Was this path mentioned somewhere in a docs? It changed but I don't see any change to where they were created (nor in the documentation). It'd be probably nice to mention it somewhere.

bpradipt commented:
@Amulyam24 can you please rebase the PR.

Amulyam24 commented:
PR is ready for review, PTAL!

Amulyam24 commented:
Can this PR be merged?

stevenhorsman commented:
> Can this PR be merged?

You didn't respond to Lukas' comment: https://github.com/confidential-containers/cloud-api-adaptor/pull/2366#pullrequestreview-2753415745
Has that been resolved?

Amulyam24 commented:
> > Can this PR be merged?
> 
> You didn't seem to respond to Lukas' comment: [#2366 (review)](https://github.com/confidential-containers/cloud-api-adaptor/pull/2366#pullrequestreview-2753415745) Has that been resolved?

Yes, I addressed all the comments and responded in the [thread](https://github.com/confidential-containers/cloud-api-adaptor/pull/2366#discussion_r2048440864).
